repos/system_upgrade/el7toel8/actors/scansaphana/tests/test_scansaphana.py

```diff
@@ -0,0 +1,177 @@
+        self.admusername = admusername
+
+    def __call__(self, *args, **kwargs):
+        assert args[0][0:3] == ['sudo', '-u', self.admusername]
+        cmd = args[0][3:]
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        p.wait()
+        return {'exit_code': p.returncode, 'stdout': p.stdout.read()}
+
+
+def test_scansaphana_get_instance_status(monkeypatch):
+    call = CallMock(ret={})
+    monkeypatch.setattr(scansaphana, 'run', call)
+
+    call.ret = {'exit_code': 3, 'stdout': ''}
+    assert scansaphana.get_instance_status('00', 'fake/control/path', 'tstadm')
+    assert call.args[0] == ['sudo', '-u', 'tstadm', 'fake/control/path', '-nr', '00', '-function', 'GetProcessList']
+
+    call.ret = {'exit_code': 4, 'stdout': ''}
+    assert not scansaphana.get_instance_status('00', 'fake/control/path', 'tstadm')
+    assert call.args[0] == ['sudo', '-u', 'tstadm', 'fake/control/path', '-nr', '00', '-function', 'GetProcessList']
+
+    call.ret = {'exit_code': 0, 'stdout': '\n'}
+    assert not scansaphana.get_instance_status('00', 'fake/control/path', 'tstadm')
+    assert call.args[0] == ['sudo', '-u', 'tstadm', 'fake/control/path', '-nr', '00', '-function', 'GetProcessList']
+
+    call.ret = {'exit_code': 0, 'stdout': ' \n' * 6}
+    assert not scansaphana.get_instance_status('00', 'fake/control/path', 'tstadm')
+    assert call.args[0] == ['sudo', '-u', 'tstadm', 'fake/control/path', '-nr', '00', '-function', 'GetProcessList']
+
+    call.ret = {'exit_code': 0, 'stdout': '\n' * 7}
+    assert scansaphana.get_instance_status('00', 'fake/control/path', 'tstadm')
+    assert call.args[0] == ['sudo', '-u', 'tstadm', 'fake/control/path', '-nr', '00', '-function', 'GetProcessList']
+
+
+def test_scansaphana_parse_manifest(monkeypatch):
+    class _mock_open(object):
+        def __init__(self, path, mode):
+            self._fp = BytesIO(SAPHANA2_MANIFEST.encode('utf-8'))
+
+        def __enter__(self):
+            return self._fp
+
+        def __exit__(self, *args, **kwargs):
+            return None
+
+    monkeypatch.setattr(scansaphana, 'open', _mock_open, False)
+    data = scansaphana.parse_manifest('yadda')
+    assert data['sapexe-version'] == '753'
+    assert data['keycaption'] == 'SAP HANA DATABASE'
+
+
+def test_scansaphana_search_saphana_databases(monkeypatch, tmpdir):
+    base_path = tmpdir
+    monkeypatch.setattr(scansaphana, 'HANA_BASE_PATH', str(base_path))
+    cur_path = base_path
+    manifest_path_parts = scansaphana.HANA_MANIFEST_PATH.split('/')
+    for part in manifest_path_parts[0:-1]:
+        cur_path = cur_path.join(part)
+        cur_path.mkdir()
+
+    linkpath = base_path.join(manifest_path_parts[0])
+
+    manifest = cur_path.join(manifest_path_parts[-1])
+    manifest.write_text(unicode_type(SAPHANA2_MANIFEST), encoding='utf-8')
+
+    sapcontrol = cur_path.join(scansaphana.HANA_SAPCONTROL_PATH.split('/')[-1])
+    sapcontrol.write_text(unicode_type('''#!/bin/bash
+echo 'Line1'
```

Comment:
could you please elaborate on what we are trying to achieve with this test?

Reply:
This was just a fake output. I can replace it with real output examples.